snp_motion_planning/src/planning_server.cpp

```diff
@@ -576,13 +616,67 @@ class PlanningServer
 
     RCLCPP_INFO_STREAM(node_->get_logger(), res->message);
   }
+  void freespaceMotionPlanCallback(const snp_msgs::srv::GenerateFreespaceMotionPlan::Request::SharedPtr req,
+                                   snp_msgs::srv::GenerateFreespaceMotionPlan::Response::SharedPtr res)
+  {
+    try
+    {
+      RCLCPP_INFO_STREAM(node_->get_logger(), "Received freespace motion planning request");
+
+      tesseract_common::ManipulatorInfo manip_info;
+      manip_info.manipulator = req->motion_group;
+      manip_info.tcp_frame = req->tcp_frame;
+      manip_info.working_frame = req->mesh_frame;
+
+      tesseract_planning::CompositeInstruction freespace_program(
+          PROFILE, tesseract_planning::CompositeInstructionOrder::ORDERED, manip_info);
+
+      tesseract_planning::JointWaypoint wp1 = rosJointStateToJointWaypoint(req->js1);
+      tesseract_planning::JointWaypoint wp2 = rosJointStateToJointWaypoint(req->js2);
+
+      // Define a freespace move to the first waypoint
+      freespace_program.appendMoveInstruction(
+          tesseract_planning::MoveInstruction(tesseract_planning::JointWaypointPoly{ wp1 },
+                                              tesseract_planning::MoveInstructionType::FREESPACE, PROFILE, manip_info));
+
+      // Define a freespace move to the second waypoint
+      freespace_program.appendMoveInstruction(
+          tesseract_planning::MoveInstruction(tesseract_planning::JointWaypointPoly{ wp2 },
+                                              tesseract_planning::MoveInstructionType::FREESPACE, PROFILE, manip_info));
+
+      // Add the scan link to the planning environment
+      addScanLink(req->mesh_filename, req->mesh_frame);
```

Comment:
We should make this optional. In the case that you want to just plan to some arbitrary position like a home state or something and you don't have any scan mesh to add yet.

Reply:
It is now optional. No mesh will be added if the filename is left empty.